cla_public/apps/checker/forms.py

```diff
@@ -145,8 +192,32 @@ def value_or_zero(field, dependant_field):
                 and self.partner_is_self_employed.data == YES:
             payload['partner'] = {'income': {
                                   'self_employed': self.partner_is_self_employed.data}}
+
+        # need to delete properties if user changes to no property
+        # also need to save a blank property now if user selects owns_property
+        properties_data = PropertiesForm.get_session_as_api_payload() if \
+            self.require_properties else PropertiesForm.get_zero_api_payload()
+        recursive_dict_update(payload, properties_data)
+
+        # set savings to zero/null depending on has_savings/has_valuables
+        savings_data = SavingsForm.get_session_as_api_payload() if \
+            self.require_savings else SavingsForm.get_zero_api_payload()
+        recursive_dict_update(payload, savings_data)
+
+        # add null/session income and set to 0 if on passported benefits later
+        income = IncomeForm().get_session_as_api_payload()
+        recursive_dict_update(payload, income)
```

Comment:
Comments mean the code is not clear enough and there's repetition of `Form.get_session_as_api_payload` and `recursive_dict_update`. I think it would be clearer abstracted into a function, eg:
```python
def update_payload(form_class, cond=True):
    if cond:
        form_data = form_class.get_session_as_api_payload()
    else:
        form_data = form_class.get_zero_api_payload()
    recursive_dict_update(payload, form_data)

update_payload(PropertiesForm, cond=self.require_properties)
update_payload(SavingsForm, cond=self.require_savings)
update_payload(IncomeForm)
```